src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/DataTableCaption.test.tsx

```diff
@@ -0,0 +1,378 @@
+import React from 'react';
+import {
+  CategoryFilter,
+  Indicator,
+  LocationFilter,
+  TimePeriodFilter,
+} from '@common/modules/table-tool/types/filters';
+import { FullTableMeta } from '@common/modules/table-tool/types/fullTable';
+import { render, screen, fireEvent } from '@testing-library/react';
+import DataTableCaption from '../DataTableCaption';
+
+describe('DataTableCaption', () => {
+  const generateTableTitleBasicInput: FullTableMeta = {
+    geoJsonAvailable: false,
+    publicationName: 'Pupil absence in schools in England',
+    subjectName: 'Absence by characteristic',
+    footnotes: [],
+    boundaryLevels: [],
+    filters: {
+      Characteristic: {
+        name: 'characteristic',
+        options: [
+          new CategoryFilter({
+            value: 'total',
+            label: 'Total',
+            group: 'Gender',
+            category: 'Characteristic',
+          }),
+        ],
+      },
+    },
+    indicators: [
+      new Indicator({
+        label: 'Authorised absence rate',
+        value: 'authAbsRate',
+        unit: '%',
+        name: 'sess_authorised_percent',
+      }),
+    ],
+    locations: [
+      new LocationFilter({
+        value: 'E92000001',
+        label: 'England',
+        level: 'country',
+      }),
+    ],
+    timePeriodRange: [
+      new TimePeriodFilter({
+        code: 'AY',
+        year: 2015,
+        label: '2015/16',
+        order: 0,
+      }),
+    ],
+  };
+
+  test('with given title', () => {
+    const { container } = render(
+      <DataTableCaption
+        id="DataTableCaption"
+        title="Test title"
+        {...generateTableTitleBasicInput}
+      />,
+    );
+
+    expect(screen.queryByText('Test title')).not.toBeNull();
+    expect(container).not.toHaveTextContent('full table title');
+  });
+
+  test('with multiple filters labelled "Total"', () => {
+    const props = {
+      ...generateTableTitleBasicInput,
+      filters: {
+        ...generateTableTitleBasicInput.filters,
+        'School Type': {
+          name: 'school_type',
+          options: [
+            new CategoryFilter({
+              value: 'total',
+              label: 'Total',
+              category: 'School Type',
+            }),
+          ],
+        },
+      },
+    };
+    const { container } = render(
+      <DataTableCaption id="DataTableCaption" {...props} />,
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
```

Comment:
When I said inline snapshot, I didn't mean this.

Swap the data table caption `id` to a test id. Use this in combination with the `screen.getByTestId` to do:

```javascript
expect(screen.getByTestId('dataTableCaption')).toMatchInlineSnapshot(...)
```

You'll also need to update the UI tests to use this as well, as we can use the `data-testid` instead.